File: src/main/java/com/severett/kotlinintro/model/Cat.java
package com.severett.kotlinintro.model;

import jakarta.persistence.Entity;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@NoArgsConstructor
@Getter
@Setter
public class Cat extends Pet {
    private static final int HASHCODE = Cat.class.hashCode();

    private double clawLength;

    public Cat(int id, String name, double clawLength) {
        super(id, name);
        this.clawLength = clawLength;
    }

    @Override
    public String getSound() {
        return "Meows";
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Cat)) return false;
        return getId() == ((Cat) o).getId();
    }

    @Override
    public int hashCode() {
        return HASHCODE;
    }
}

File: src/main/kotlin/com/severett/kotlinintro/model/Cat.kt
package com.severett.kotlinintro.model

import jakarta.persistence.Entity

private val HASHCODE: Int = Cat::javaClass.hashCode()

@Entity
class Cat(id: Int = 0, name: String = "", var clawLength: Double = 0.0) : Pet(id, name) {
    override val sound: String
        get() = "Meows"

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Cat) return false
        return id == other.id
    }

    override fun hashCode() = HASHCODE
}
